includes/class-wpcom-legacy-redirector.php

```diff
@@ -443,4 +444,31 @@ public static function vip_legacy_redirect_parent_id( $post ) {
 			}
 		}
 	}
+
+	/**
+ 	 * Returns current URL's relative path including the query string. Optionally add or remove trailig slash to get variant URL.
+ 	 * @param bool | $trailing_slash_toggle If TRUE, will return URL toggling the trailing slash on/off (i.e add slash if not present).
+ 	 * @return string | Will return the current URL path based via $_SERVER.
+ 	 */
+	static function get_current_url_path( $trailing_slash_toggle = false ) {
+
+		$url = parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH );
```

Comment:
Those are solid recommendations, I will improve it. Wanted to initially keep the code changes minimal since this was existing/working functionality then take a second pass time permitting.

Reply:
$_SERVER['REQUEST_URI' is filled by wp_fix_server_vars() by default so it won't be empty. I'll proceed to check just in case.

Reply:
@GaryJones Why is it necessary to manually parse and re-add the query string to the URL? REQUEST_URI already includes the relative path including query_strings. i.e all we need is ... 

`$url = esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) );`